File: sdk/kotlin/src/main/kotlin/vn/hitechcloud/sdk/Ipam.kt
package vn.hitechcloud.sdk

/** HiTechCloudIPAM: IP addresses, subnets, reverse DNS */
class IpamResource(private val httpClient: HttpClient) {
    /** GET /api/service/{id}/htcipam/ips - List IP addresses */
    fun listIps(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/htcipam/ips", params)
    }

    /** GET /api/service/{id}/htcipam/subnets - List subnets */
    fun listSubnets(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/htcipam/subnets", params)
    }

    /** GET /api/service/{id}/htcipam/rdns - List reverse DNS records */
    fun listRdns(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.get("/api/service/{id}/htcipam/rdns", params)
    }

    /** POST /api/service/{id}/htcipam/rdns - Set reverse DNS */
    fun setRdns(params: Map<String, Any?> = emptyMap()): Map<String, Any> {
        return httpClient.post("/api/service/{id}/htcipam/rdns", params)
    }

}

File: sdk/swift/Sources/HiTechCloudSDK/Ipam.swift
import Foundation

/// HiTechCloudIPAM: IP addresses, subnets, reverse DNS
public class IpamResource {
    private let httpClient: HTTPClient
    public init(httpClient: HTTPClient) { self.httpClient = httpClient }

    /// GET /api/service/{id}/htcipam/ips - List IP addresses
    public func listIps(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/htcipam/ips", params: params)
    }

    /// GET /api/service/{id}/htcipam/subnets - List subnets
    public func listSubnets(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/htcipam/subnets", params: params)
    }

    /// GET /api/service/{id}/htcipam/rdns - List reverse DNS records
    public func listRdns(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.get("/api/service/{id}/htcipam/rdns", params: params)
    }

    /// POST /api/service/{id}/htcipam/rdns - Set reverse DNS
    public func setRdns(params: [String: Any?] = [:]) throws -> [String: Any] {
        return try httpClient.post("/api/service/{id}/htcipam/rdns", params: params)
    }

}
